src/context/cart-context.js

```diff
@@ -0,0 +1,139 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import { useLogin } from "../context/index";
+import axios from "axios";
+const CartContext = createContext();
+const useCart = () => useContext(CartContext);
+
+const CartProvider = ({ children }) => {
+  const { userData } = useLogin();
+  const [cartItems, setCartItems] = useState([]);
+
+  const getCartItems = async () => {
```

Comment:
maybe extract all these functions into a separate file to improve code readability.